File: ruby/src/Behavioral/Template_Method/DvdTitleInfo.rb
#DvdTitleInfo - one of three concrete templates

require "Behavioral/Template_Method/TitleInfo"

class DvdTitleInfo < TitleInfo
	def initialize(titleName,star,encodingRegion)
 		@titleName = titleName
		@star = star
		@encodingRegion = encodingRegion
	end

	def getTitleBlurb()
 		"DVD: #{getTitleName}, starring #{@star}"
	end

	def getDvdEncodingRegionInfo()
		", encoding region: #{@encodingRegion}"
	end

end

File: csharp/src/Behavioral/Template_Method/DvdTitleInfo.cs
namespace Behavioral.Template_Method {


//DvdTitleInfo.java - one of three concrete templates

public class DvdTitleInfo : TitleInfo {  
   private string star;
   private char encodingRegion;
    
   public DvdTitleInfo(string titleName, 
                       string star, 
                       char encodingRegion) {
       this.setTitleName(titleName);
       this.setStar(star);
       this.setEncodingRegion(encodingRegion);
   }    
   
   public void setStar(string starIn) {
       this.star = starIn;
   }
   public string getStar() {
       return this.star;
   }
   public void setEncodingRegion(char encodingRegionIn) {
       this.encodingRegion = encodingRegionIn;
   }
   public char getEncodingRegion() {
       return this.encodingRegion;
   }
   
   override public string getTitleBlurb() {
       return ("DVD: " + this.getTitleName() + 
               ", starring " + this.getStar());
   }
   
   public new string getDvdEncodingRegionInfo() {
       return (", encoding region: " + this.getEncodingRegion());
   }
}

}
